modules/light-clients/06-solomachine/solomachine_test.go

```diff
@@ -46,16 +46,24 @@ func TestSoloMachineTestSuite(t *testing.T) {
 	suite.Run(t, new(SoloMachineTestSuite))
 }
 
-func (suite *SoloMachineTestSuite) TestConnectionHandshake() {
+func (suite *SoloMachineTestSuite) Setup() {
 	clientID := suite.solomachine.CreateClient(suite.chainA)
 
 	connectionID := suite.solomachine.ConnOpenInit(suite.chainA, clientID)
 
-	// open try is not necessary as the solo machine implementation is mock'd
+	// open try is not necessary as the solo machine implementation is mocked
 
 	suite.solomachine.ConnOpenAck(suite.chainA, clientID, connectionID)
 
-	// open confirm is not necessary as the solo machine implementation is mock'd
+	// open confirm is not necessary as the solo machine implementation is mocked
+
+	channelID := suite.solomachine.ChanOpenInit(suite.chainA, connectionID)
+
+	// open try is not necessary as the solo machine implementation is mocked
+
+	suite.solomachine.ChanOpenAck(suite.chainA, channelID)
+
+	// open confirm is not necessary as the solo machine implementation is mocked
```

Comment:
I almost think we should still call the mocked function just for sake of clarity in what actually happens.

Reply:
This is kind of tricky, what we are really doing here is just pretending the solomachine endpoints are being called properly. If we wanted to mock it out fully we could maybe compose the `Solomachine` type in `testing` with a `ClientState` and assign it to the `07-tendermint` client from `suite.chainB` and keep it up to date.
This would track the state of chainA for the solomachine but it may be a bit hacky as we still have this chainB tendermint chain. What we really want is `chainA(06-solomachine client)` <-> `solomachine(07-tendermint client)`

I think it would be interesting to attempt to mock out the flow fully but I think there'd be a bit of work involved.

Reply:
ah okay I didn't realise this was so involved, I'm happy enough to leave it as a comment now if this is the case! 👍